integration/spark/src/main/scala/org/apache/spark/sql/CarbonDictionaryDecoder.scala

```diff
@@ -158,7 +158,8 @@ case class CarbonDictionaryDecoder(
         val carbonTable = relation.carbonRelation.carbonRelation.metaData.carbonTable
         (carbonTable.getFactTableName, carbonTable.getAbsoluteTableIdentifier)
       }.toMap
-      val recorder = new QueryStatisticsRecorder(queryId)
+
+      val recorder = CarbonTimeStatisticsFactory.createRecorder(queryId);
```

Comment:
This should be task/executor side recorder, right? Why not name it createTaskRecorder or createExecutorRecorder